File: src/main/java/quoters/MainService.java
package quoters;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import javax.annotation.PostConstruct;

/**
 * @author Evgeny Borisov
 */
@Service
public class MainService {

    @Autowired
    private QuoterAggregator quoterAggregator;

    @PostConstruct
    public void doWork(){
        quoterAggregator.sayAllQuotes();
    }
}

File: src/main/java/scala_implicit_examples/scala_to_java_convertors/MainService.scala
package scala_implicit_examples.scala_to_java_convertors

import javax.annotation.PostConstruct
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import collection.JavaConverters._
/**
  * @author Evgeny Borisov
  */
@Component
class MainService {

  @Autowired
  var validators:java.util.List[Validator]=null


  @PostConstruct
  def printAllValidations():Unit={
    validators.asScala.foreach(_.validate())


  }

}
